### DISCUSSION TITLE
Issue with text layer rendering: pdfjs v3.3.122

### DISCUSSION THREAD

**Discussion by marija-popovic** [mozilla/pdf.js#18262] [Q&A]
Since upgrading from pdfjs **v2.5.207** to pdfjs **v3.3.122** I have had issues with rendering a text layer of pdfs. Previously, for this purpose, class _TextLayerBuilder_ was used and it was working correctly.

This is the code that was used:

```
const textLayer = new TextLayerBuilder({
        eventBus: new Proxy({}, { get: () => () => {} }),
        textLayerDiv: this.textLayerElement.nativeElement,
        pageIndex: this.page.pageIndex,
        viewport,
});

textLayer.setTextContent(this.textContent);
textLayer.render();
```

Now, the constructor of _TextLayerBuilder_ doesn't accept the same parameters as before and it is not clear how the same behavior should be achieved with these changes. Instead, I tried to use _renderTextLayer_ function:

```
const renderTextLayerTask = renderTextLayer({
        textContentSource: this.textContent,
        container: this.textLayerElement.nativeElement,
        viewport,
        textDivs: [],
});
```

but it resulted in having issues with the alignment and font size of the pdf text, for example: 

![small-pdf](https://github.com/mozilla/pdf.js/assets/89454949/e749a1c4-7a69-4a50-bfd3-f6de123c29a1)

I would appreciate if some instructions for overcoming this issue could be provided, either with the usage of _TextLayerBuilder_ or _renderTextLayer_.

**Comment by Daschmi** [comment]
Same Problem with scaled DIV.

pdfjs-dist@5.4.530

```js
		const textLayer = new TextLayerBuilder({
			pdfPage: page,
			viewport: scaledViewport
		});

		textLayer.div = textLayerDiv;

		const textContent = await page.getTextContent();

		await textLayer.render({
			viewport: scaledViewport
		});
```

<img width="738" height="497" alt="image" src="https://github.com/user-attachments/assets/d6459b06-89cb-44b7-b7dd-dcf1f10f3d61" />
